### PR TITLE
DBG: Add debug logs to track down segfault

### PR DESCRIPTION
**Pull request by effigies**
Adjunct to #2409.

### CODE REVIEW

File: fmriprep/utils/sentry.py [hunk 1]
```diff
@@ -52,7 +52,8 @@ def sentry_setup():
     sentry_sdk.init("https://d5a16b0c38d84d1584dfc93b9fb1ade6@sentry.io/1137693",
                     release=release,
                     environment=environment,
-                    before_send=before_send)
+                    before_send=before_send,
+                    shutdown_timeout=0)
```
Comment (mgxd) [inline fmriprep/utils/sentry.py:56]: perhaps we can get something useful directly from sentry?
```suggestion
                    shutdown_timeout=0,
                    debug=True)
```

Comment (effigies) [reply fmriprep/utils/sentry.py]: Good call. Added.

**Comment by effigies** [comment]
With `AtexitIntegration`:

```
210611-01:31:02,263 nipype.workflow IMPORTANT:
	 Closing Sentry client pre-shutdown...
 [sentry] DEBUG: Flushing HTTP transport
 [sentry] DEBUG: background worker got flush request
 [sentry] DEBUG: background worker flushed
 [sentry] DEBUG: Killing HTTP transport
 [sentry] DEBUG: background worker got kill request
210611-01:31:02,264 nipype.workflow IMPORTANT:
	 Exiting...
 [sentry] DEBUG: Killing HTTP transport
 [sentry] DEBUG: background worker got kill request
 [sentry] DEBUG: atexit: got shutdown signal
 [sentry] DEBUG: atexit: shutting down client
Fatal Python error: Segmentation fault

Thread 0x00007f33f2e19700 (most recent call first):
  File "/usr/local/miniconda/lib/python3.8/site-packages/sentry_sdk/worker.py", line 126 in _target
  File "/usr/local/miniconda/lib/python3.8/threading.py", line 870 in run
  File "/usr/local/miniconda/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/local/miniconda/lib/python3.8/threading.py", line 890 in _bootstrap

Current thread 0x00007f3418bc6740 (most recent call first):
<no Python frame>
```

With our own `atexit` teardown:

```
210611-03:13:17,16 nipype.workflow IMPORTANT:
	 Closing Sentry client pre-shutdown...
 [sentry] DEBUG: Flushing HTTP transport
 [sentry] DEBUG: background worker got flush request
 [sentry] DEBUG: background worker flushed
 [sentry] DEBUG: Killing HTTP transport
 [sentry] DEBUG: background worker got kill request
Fatal Python error: Segmentation fault

Thread 0x00007f143a413700 (most recent call first):
  File "/usr/local/miniconda/lib/python3.8/threading.py", line 302 in wait
  File "/usr/local/miniconda/lib/python3.8/site-packages/sentry_sdk/_queue.py", line 180 in get
  File "/usr/local/miniconda/lib/python3.8/site-packages/sentry_sdk/worker.py", line 126 in _target
  File "/usr/local/miniconda/lib/python3.8/threading.py", line 870 in run
  File "/usr/local/miniconda/lib/python3.8/threading.py", line 932 in _bootstrap_inner
  File "/usr/local/miniconda/lib/python3.8/threading.py", line 890 in _bootstrap

Current thread 0x00007f14601c0740 (most recent call first):
<no Python frame>
```

Seems it's a thread spawned by sentry that's causing problems, regardless of how it's killed. Have a hunch that it's a result of https://github.com/getsentry/sentry-python/pull/994, so pinning <1.0.0 (see https://github.com/getsentry/sentry-python/releases/tag/1.0.0).

**Comment by effigies** [comment]
Giving up for now. Will leave open in case someone wants to try to pick it up, but feel free to close it.

**Comment by effigies** [comment]
@mgxd It's not Sentry... I think it may be nipype.

**Comment by mgxd** [comment]
this can be closed since #2409 is in

**Pull request closed without merging**